sabnzbd/sorting.py

```diff
@@ -1021,62 +732,48 @@ def path_subst(path, mapping):
                     n += len(key) - 1
                     result = value
                     break
-        newpath.append(result)
+        if result:
+            newpath.append(result)
         n += 1
     return "".join(newpath)
 
 
-def get_titles(nzo: NzbObject, match, name, titleing=False):
-    """The title will be the part before the match
-    Clean it up and title() it
-
-    ''.title() isn't very good under python so this contains
-    a lot of little hacks to make it better and for more control
-    """
+def get_titles(
+    nzo: Optional[NzbObject], guess: MatchesDict, jobname: str, titleing: bool = False
+) -> Tuple[str, str, str]:
+    """Get the title from NZB metadata or jobname, and return it in various formats. Formatting
+    mostly deals with working around quirks of Python's str.title(). NZB metadata is used as-is,
+    further processing done only for info obtained from guessit or the jobname."""
+    title = ""
     if nzo:
+        # Fetch NZB metadata
         title = nzo.nzo_info.get("propername")
```

Comment:
I think we should get rid of this, or at least not rely on this as the only input, what do you think>

Reply:
Why? This would be set by the uploader or indexer, both of whom sab has no choice but to trust when it comes to the content of the nzb. No reason not to extend that to the proper naming.